fastly/tls_mutual_authentication.go

```diff
@@ -0,0 +1,212 @@
+	ID string
+	// Include is a comma-separated list of related objects to include (optional).
+	Include string
+}
+
+// GetTLSMutualAuthentication retrieves the specified resource.
+func (c *Client) GetTLSMutualAuthentication(i *GetTLSMutualAuthenticationInput) (*TLSMutualAuthentication, error) {
+	if i.ID == "" {
+		return nil, ErrMissingID
+	}
+
+	var ro *RequestOptions
+	if i.Include != "" {
+		ro = &RequestOptions{
+			Params: map[string]string{
+				"include": i.Include,
+			},
+		}
+	}
+
+	p := fmt.Sprintf("/tls/mutual_authentications/%s", i.ID)
+
+	r, err := c.Get(p, ro)
+	if err != nil {
+		return nil, err
+	}
+
+	var o TLSMutualAuthentication
+	if err := jsonapi.UnmarshalPayload(r.Body, &o); err != nil {
+		return nil, err
+	}
+
+	return &o, nil
+}
+
+// CreateTLSMutualAuthenticationInput is used as input to the CreateTLSMutualAuthentication function.
+type CreateTLSMutualAuthenticationInput struct {
+	// CertBundle is one or more certificates. Enter each individual certificate blob on a new line. Must be PEM-formatted (required).
+	CertBundle string `jsonapi:"attr,cert_bundle"`
+	// ID should not be set (it's internally used to help marshal the JSONAPI request data).
+	ID string `jsonapi:"primary,mutual_authentication"`
+	// Name is a custom name for your mutual authentication (optional).
+	Name string `jsonapi:"attr,name,omitempty"`
+}
+
+// CreateTLSMutualAuthentication creates a new resource.
+func (c *Client) CreateTLSMutualAuthentication(i *CreateTLSMutualAuthenticationInput) (*TLSMutualAuthentication, error) {
+	if i.CertBundle == "" {
+		return nil, ErrMissingCertBundle
+	}
+
+	p := "/tls/mutual_authentications"
+
+	r, err := c.PostJSONAPI(p, i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var o TLSMutualAuthentication
+	if err := jsonapi.UnmarshalPayload(r.Body, &o); err != nil {
+		return nil, err
+	}
+
+	return &o, nil
+}
+
+// UpdateTLSMutualAuthenticationInput is used as input to the UpdateTLSMutualAuthentication function.
+type UpdateTLSMutualAuthenticationInput struct {
+	// CertBundle is one or more certificates. Enter each individual certificate blob on a new line. Must be PEM-formatted (required).
+	CertBundle string `jsonapi:"attr,cert_bundle"`
+	// Enforced determines whether Mutual TLS will fail closed (enforced) or fail open.
+	Enforced bool `jsonapi:"attr,enforced"`
+	// ID is an alphanumeric string identifying a mutual authentication (required).
+	ID string `jsonapi:"primary,mutual_authentication"`
+	// Name is a custom name for your mutual authentication (optional).
+	Name string `jsonapi:"attr,name,omitempty"`
+}
+
+// UpdateTLSMutualAuthentication updates the specified resource.
+//
+// By using this endpoint, the original certificate will cease to be used for future TLS handshakes.
+// Thus, only SAN entries that appear in the replacement certificate will become TLS enabled.
+// Any SAN entries that are missing in the replacement certificate will become disabled.
```

Comment:
I don't think this text applies for this API.  If I am understanding this API, the cert bundle being uploaded is the set of CA certificates used to validate certificates being presented by TLS clients during the handshake.  If that's the case, SANs don't apply here.